### PR TITLE
Go & tooling upgrades

### PR DESCRIPTION
**Pull request by manicminer**
- Update to Go 1.19.2
- Common go version in workflows
- Update golangci-lint action & version
- Update setup-go action
- Configure tests for a B2C tenant

### CODE REVIEW

File: .github/workflows/auth-tests.yml [hunk 1]
```diff
@@ -22,18 +22,17 @@ jobs:
...
     steps:
-      - name: Azure CLI login
-        run: az login --allow-no-subscriptions --output none --service-principal --tenant "${TENANT_ID}" --username "${CLIENT_ID}" --password "${CLIENT_SECRET}"
+      - run: az login --allow-no-subscriptions --output none --service-principal --tenant "${TENANT_ID}" --username "${CLIENT_ID}" --password "${CLIENT_SECRET}"
 
-      - name: Install Go
-        uses: actions/setup-go@v2
-        with:
-          go-version: 1.17.6
+      - uses: actions/checkout@v2
```
Comment (tombuildsstuff) [inline .github/workflows/auth-tests.yml:27]: it'd be worth making this `v3` if possible, `v2`'s been unreliable of late

Comment (manicminer) [reply .github/workflows/auth-tests.yml:27]: Good catch, thanks 👍

**Review by jackofallops — APPROVED** [review]
One minor question / suggestion, otherwise LGTM 👍

File: .github/workflows/auth-tests.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (manicminer) [reply .github/workflows/auth-tests.yml]: Ah good spot, thanks!

**Pull request merged**